packages/cardpay-reward-api/tests/test_reward_api.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import pytest
+from cardpay_reward_api.database import Base, get_db
+from cardpay_reward_api.indexer import Indexer
+from cardpay_reward_api.main import app
+from cardpay_reward_api.models import Root
+from fastapi.testclient import TestClient
+
+from .fixtures import engine, override_get_db, roots_for_program
+from .mocks import reward_programs
+from .utils import (check_duplicates_for_proofs, check_duplicates_for_roots,
+                    validate_proof_response_fields)
+
+REWARDS_BUCKET = "tests/resources"
+ENVIRONMENT = "local"
+
+app.dependency_overrides[get_db] = override_get_db
+
+client = TestClient(app)
+
+
+@pytest.fixture(scope="module")
+def monkeymodule():
+    """
+    We need to use monkeypatch when patching a fixture
+    But the issue is that, we will always receive a ScopeMismatch error
+    when wanting the fixture to run once in a specific scope (in this case a module)
+    https://stackoverflow.com/questions/53963822/python-monkeypatch-setattr-with-pytest-fixture-at-module-scope
+    """
+    from _pytest.monkeypatch import MonkeyPatch
+
+    mpatch = MonkeyPatch()
+    yield mpatch
+    mpatch.undo()
```

Comment:
One of the issuees with monkeypatch is that I could not get it to work with scoped fixtures very well. So, I had to patch the monkeypatch st it is scoped. That way I can instantiate the fixture for a single file and not run it every test